src/redux/books/booksSlice.js

```diff
@@ -1,42 +1,73 @@
 import { createSlice } from '@reduxjs/toolkit';
+import { addNewBook, fetchBooks, removeBook } from '../apis';
 
 const initialState = {
-  books: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  isLoading: false,
+  error: null,
 };
 
+// create slice
 const booksSlice = createSlice({
   name: 'bookstore',
   initialState,
-  reducers: {
-    addBook: (state, action) => {
-      const newBook = action.payload;
-      state.books.push(newBook);
-    },
-    removeBook: (state, action) => {
-      const itemId = action.payload;
-      state.books = state.books.filter((book) => book.item_id !== itemId);
-    },
+  reducers: {},
+
+  // handle states of promises
+  extraReducers: (builder) => {
+    builder
+      // fetch books
+      .addCase(fetchBooks.pending, (state) => {
+        state.isLoading = true;
+      })
+      .addCase(fetchBooks.fulfilled, (state, { payload }) => {
+        state.isLoading = false;
+        const result = [];
+        Object.entries(payload).forEach(([item, itemsArray]) => {
+          itemsArray.forEach((itemData) => {
+            const newItem = {
+              item_id: item,
+              author: itemData.author,
+              title: itemData.title,
+              category: itemData.category,
```

Comment:
- [ ] As a result of the `item_id` not being unique, sometimes it is impossible to create a new book record. I looked at your `generateUniqueId` function in your `navigationsLinks.js` file, and I noticed that you are trying to generate a unique ID based off of a `counter` variable that is not persisted. 

The problem with this approach is that once the page gets reloaded, `counter` becomes 0. Now, any other book record that you attempt to create would have an existing ID and that throws an error. 
I suggest you use some library like `uuid` instead to avoid such bugs 🐛
![image](https://github.com/til2to/bookstore/assets/46839548/7b7dbcd9-ca49-446a-8285-2990457f6eae)